src/algebra/group_ring_action.lean

```diff
@@ -65,6 +65,18 @@ def mul_semiring_action.to_ring_equiv [mul_semiring_action G R] (x : G) : R ≃+
 { .. distrib_mul_action.to_add_equiv R x,
   .. mul_semiring_action.to_ring_hom G R x }
 
+/-- A multiplicative action of `N` on `R` and a group homomorphism `K →* N` induce
+a multiplicative action of `K` on `R`. -/
+def mul_semiring_action.comp_hom (g : K →* N) [mul_semiring_action N R] :
+  mul_semiring_action K R :=
+{ smul := has_smul.comp.smul g,
+  one_smul := λ a, by simp only [has_smul.comp.smul, map_one, one_smul],
+  smul_zero:= λ n, by simp only [has_smul.comp.smul, smul_zero],
+  smul_one := λ x, by simp [has_smul.comp.smul, smul_one],
+  smul_mul := λ n x y, by simp [has_smul.comp.smul, smul_mul'],
+  smul_add := λ n x y, by simp [has_smul.comp.smul, smul_add],
+  mul_smul := λ x y a, by simp only [has_smul.comp.smul, map_mul,mul_smul (g x) (g y) a] }
```

Comment:
You said you got errors on @eric-wieser's suggestion, and I think that is because there were some missing arguments. This should work:
```suggestion
{ smul := has_smul.comp.smul g,
  ..distrib_mul_action.comp_hom R g,
  ..mul_distrib_mul_action.comp_hom R g }
```

Reply:
Great, thank you very much. It definitely looks better :)